### PR TITLE
[blog][hopl] Tracing JITs

### PR DESCRIPTION
**Pull request by mhyee**
Not quite finished.

I'm still typesetting my lecture notes, and it'll probably take a few more days. But @bennn asked me to prepare a blog post for now.

It seems a bit silly to publish this if **both** the discussion notes and lecture notes are missing. I'd rather wait until the lecture notes are done, but in the meantime, it's probably better to merge the discussion notes (https://github.com/nuprl/hopl-s2017/pull/7) (and maybe commit @gasche's version as well). I can update the blog post later, when the lecture notes are done.

Thoughts?

### CODE REVIEW

File: blog/_src/posts/2017-03-10-tracing-jits-for-dynamic-languages.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+    Title: Tracing JITs for Dynamic Languages
+    Date: 2017-03-10T15:54:39
+    Tags: HOPL, by Ming-Ho Yee
+
+<!-- more -->
+
+Traditional JIT (just-in-time) compilers are method-based: they compile "hot"
+(i.e. frequently executed) methods to native code. An alternative is trace-based
+or tracing JITs, where the compilation unit is a (hot) sequence of instructions.
+Typically, such sequences of instructions correspond to a loops, where programs
```
Comment (bennn) [inline blog/_src/posts/2017-03-10-tracing-jits-for-dynamic-languages.md]: to a loops => to loops

**Comment by mhyee** [comment]
I'm incorporating the discussions (with some editing) into my version of the notes.

If we're publishing this blog post tomorrow, then it's probably best if you merged the PR with Gabriel's and your notes. The discussions might be duplicated, but I think that's OK.

The alternative would be to publish the blog post without any accompanying notes, or to wait until the lecture notes are done.

**Comment by bennn** [comment]
Can you have the notes by next Wednesday? If so, then lets wait for that (and not merge the other notes).

**Comment by mhyee** [comment]
OK, I'll get the notes done by then. (I'll also update the date/links in the blog post, and fix the typo.)

**Comment by mhyee** [comment]
Updated and ready to publish (unless there are last minute comments). I'm linking to the GitHub PDF because that's the canonical version.

**Comment by bennn** [comment]
Looks good, thanks for updating the date!

**Pull request merged**